dbmigrate-lib/src/drivers/sqlcipher.rs

```diff
@@ -0,0 +1,125 @@
+use url::Url;
+
+use sqlcipher_client::Connection;
+
+use super::Driver;
+use errors::{Result, ResultExt};
+
+const SQLCIPHER_DEFAULT_CIPHER: &'static str = "AES-256-CBC";
+
+#[derive(Debug)]
+pub struct Sqlcipher {
+    conn: Connection,
+}
+
+impl Sqlcipher {
+    pub fn new(url: &str) -> Result<Sqlcipher> {
+        // Expect a URL in the form:
+        // sqlcipher://user:password@host/path/to/file.db
+        //
+        // The user and host components are not used.
+        // The password is used with the PRAGMA KEY statement.
+        let parsedurl = Url::parse(url)
+            .chain_err(|| format!("Invalid SQLCipher URL: {}", url))?;
+        let path = parsedurl.path();
+        let conn = Connection::open(path)?;
+        let sqlcipher = Sqlcipher { conn: conn };
+        match parsedurl.password() {
+            Some(password) => {
+                println!("Password: {:?}", password);
+                sqlcipher.unlock(password)?;
+                sqlcipher.verify_cipher()?;
+                sqlcipher.ensure_migration_table_exists();
+            }
+            None => bail!("No password found in URL"),
+        }
+        Ok(sqlcipher)
+    }
+
+    fn unlock(&self, password: &str) -> Result<()> {
+        let pragmakey = format!("PRAGMA KEY = '{}'", password);
+        match self.conn.execute(&pragmakey, &[]) {
+            Ok(rows) => assert!(rows == 0),
+            Err(err_) => bail!("Failed to set database key: {}", err_),
+        };
+        Ok(())
+    }
+
+    fn verify_cipher(&self) -> Result<()> {
```

Comment:
Having read the SQLCipher API documentation again, I think I'm going to rename `verify_cipher`.  It's possible to use `PRAGMA cipher` to set the cipher used by SQLCipher.  It's generally not a good idea, but may possibly be required by some user in the future.  I'm not going to support it now, but I want to fix the API so adding support in the future is easier.